docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,386 @@
+
+### Values and Helper functions
+
+SDK also contains all essential values and helper functions that might be needed in your app like maximum tick, maximum price, calculating price impact and many others that will simplify your calculations.
+
+## Files
+
+```
+📦sdk
+ ┣ 📂contracts
+ ┃ ┣ 📂invariant
+ ┃ ┣ 📂PSP22
+ ┃ ┗ 📂wrapped-azero
+ ┣ 📂math
+ ┣ 📂src
+ ┗ 📂tests
+```
+
+### Contracts
+
+Contracts folder contains deploy-ready contracts and metadata that is used to interact with them.
+
+### Math
+
+Math contains Rust structs and data from main Invariant contract exported using WASM.
+
+### Source
+
+Everything combined in easy-to-use interface.
+
+### Tests
+
+Tests folder with tests to make sure everything works right.
+
+## Usage
+
+### Starting off
+
+```typescript
+const main = async () => {
+  const api = await initPolkadotApi(Network.Main) // initialize api, use enum to specify the network
+
+  // initialize account, you can use your own wallet by pasting mnemonic phase
+  const keyring = new Keyring({ type: 'sr25519' })
+  const account = await keyring.addFromUri('//Alice')
+}
+
+main()
+```
+
+### Initialize DEX and tokens
+
+:::note Why "n" is at the end of every number
+Notice how we use "n" at the end of every number. "n" indicates that specified value is a BigInt, number with higher precision. Read more about BigInt here: [BigInt MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).
+:::
+
+:::info Creating types
+You can create custom decimals with `toDecimal` syntax. First argument is an integer and second argument is a number of zeros. For example `toFee(3n, 2n)` will produce decimal equal to `0.03`. Read more about types here: [Types](types.md).
+:::
+
+:::warning Token sorting
+Tokens are sorted alphabetically when pool key is created, so make sure that you swap tokens in correct direction. Read more about pool keys here: [PoolKey](storage#poolkey).
+:::
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // load invariant contract
+  const invariant = await Invariant.load(api, Network.Main, INVARIANT_ADDRESS)
+
+  // load token contracts
+  const token0 = await PSP22.load(api, Network.Main, TOKEN0_ADDRESS) // token we want to give
+  const token1 = await PSP22.load(api, Network.Main, TOKEN1_ADDRESS) // token we want to receive
+
+  // set fee tier and pool key, make sure that pool with specified parameters exists
+  const feeTier = newFeeTier(toFee(1n, 2n), 1n) // fee: 0.01 = 1%, tick spacing: 1
+  const poolKey = newPoolKey(
+    token0.contract.address.toString(),
+    token1.contract.address.toString(),
+    feeTier
+  )
+
+  // make sure that you swap in correct direction,
+  // here we check if token0 (token we want to give) is tokenX
+  let xToY = isTokenX(token0.contract.address.toString(), token1.contract.address.toString())
+}
+
+main()
+```
+
+### Perform swap
```

Comment:
it's worth it to add sections with:
- create pool
- create position
- remove position
before swap section